actors/builtin/miner/miner_test.go

```diff
@@ -1404,18 +1552,64 @@ func (h *actorHarness) submitWindowPost(rt *mock.Runtime, deadline *miner.Deadli
 		}
 		rt.ExpectVerifyPoSt(vi, nil)
 	}
+	skipped := bitfield.New()
+	if postCfg != nil {
+		// expect power update
+		if !postCfg.expectedRawPowerDelta.IsZero() || !postCfg.expectedQAPowerDelta.IsZero() {
+			claim := &power.UpdateClaimedPowerParams{
+				RawByteDelta:         postCfg.expectedRawPowerDelta,
+				QualityAdjustedDelta: postCfg.expectedQAPowerDelta,
+			}
+			rt.ExpectSend(builtin.StoragePowerActorAddr, builtin.MethodsPower.UpdateClaimedPower, claim, abi.NewTokenAmount(0),
+				nil, exitcode.Ok)
+		}
+		if !postCfg.expectedPenalty.IsZero() {
+			rt.ExpectSend(builtin.BurntFundsActorAddr, builtin.MethodSend, nil, postCfg.expectedPenalty, nil, exitcode.Ok)
+		}
+		pledgeDelta := postCfg.expectedPenalty.Neg()
+		if !pledgeDelta.IsZero() {
+			rt.ExpectSend(builtin.StoragePowerActorAddr, builtin.MethodsPower.UpdatePledgeTotal, &pledgeDelta,
+				abi.NewTokenAmount(0), nil, exitcode.Ok)
+		}
+		skipped = *postCfg.skipped
+	}
 
 	params := miner.SubmitWindowedPoStParams{
 		Deadline:   deadline.Index,
 		Partitions: partitions,
 		Proofs:     proofs,
-		Skipped:    bitfield.BitField{},
+		Skipped:    skipped,
 	}
 
 	rt.Call(h.a.SubmitWindowedPoSt, &params)
 	rt.Verify()
 }
 
+func (h *actorHarness) computePartitions(rt *mock.Runtime, deadlines *miner.Deadlines, deadline *miner.DeadlineInfo) ([]*miner.SectorOnChainInfo, []uint64) {
```

Comment:
`Index` is the only use of `deadline`, so just take the index directly. That will also make it a bit clearer what this function does.

Please document it. I think it's something like "Computes  the set of all partition numbers, and all non-faulty sector infos included in those partitions, due at a single deadline".